src/target/riscv/debug_defines.h

```diff
@@ -3059,3 +3061,93 @@
 #define DMI_SERRX_DATA_OFFSET               0
 #define DMI_SERRX_DATA_LENGTH               0x20
 #define DMI_SERRX_DATA                      0xffffffffU
+enum riscv_debug_reg_ordinal {
+	DTM_IDCODE_ORDINAL,
+	DTM_DTMCS_ORDINAL,
+	DTM_DMI_ORDINAL,
+	DTM_BYPASS_ORDINAL,
+	CSR_DCSR_ORDINAL,
+	CSR_DPC_ORDINAL,
+	CSR_TSELECT_ORDINAL,
+	CSR_TDATA1_ORDINAL,
+	CSR_TDATA2_ORDINAL,
+	CSR_TDATA3_ORDINAL,
+	CSR_TINFO_ORDINAL,
+	CSR_TCONTROL_ORDINAL,
+	CSR_HCONTEXT_ORDINAL,
+	CSR_SCONTEXT_ORDINAL,
+	CSR_MCONTROL_ORDINAL,
+	CSR_MCONTROL6_ORDINAL,
+	CSR_ICOUNT_ORDINAL,
+	CSR_ITRIGGER_ORDINAL,
+	CSR_ETRIGGER_ORDINAL,
+	CSR_TMEXTTRIGGER_ORDINAL,
+	CSR_TEXTRA32_ORDINAL,
+	CSR_TEXTRA64_ORDINAL,
+	DM_DMSTATUS_ORDINAL,
+	DM_DMCONTROL_ORDINAL,
+	DM_HARTINFO_ORDINAL,
+	DM_HAWINDOWSEL_ORDINAL,
+	DM_HAWINDOW_ORDINAL,
+	DM_ABSTRACTCS_ORDINAL,
+	DM_COMMAND_ORDINAL,
+	DM_ABSTRACTAUTO_ORDINAL,
+	DM_CONFSTRPTR0_ORDINAL,
+	DM_CONFSTRPTR1_ORDINAL,
+	DM_CONFSTRPTR2_ORDINAL,
+	DM_CONFSTRPTR3_ORDINAL,
+	DM_NEXTDM_ORDINAL,
+	DM_DATA0_ORDINAL,
+	DM_PROGBUF0_ORDINAL,
+	DM_AUTHDATA_ORDINAL,
+	DM_DMCS2_ORDINAL,
+	DM_HALTSUM0_ORDINAL,
+	DM_HALTSUM1_ORDINAL,
+	DM_HALTSUM2_ORDINAL,
+	DM_HALTSUM3_ORDINAL,
+	DM_SBCS_ORDINAL,
+	DM_SBADDRESS0_ORDINAL,
+	DM_SBADDRESS1_ORDINAL,
+	DM_SBADDRESS2_ORDINAL,
+	DM_SBADDRESS3_ORDINAL,
+	DM_SBDATA0_ORDINAL,
+	DM_SBDATA1_ORDINAL,
+	DM_SBDATA2_ORDINAL,
+	DM_SBDATA3_ORDINAL,
+	SHORTNAME_ORDINAL,
+	AC_ACCESS_REGISTER_ORDINAL,
+	AC_QUICK_ACCESS_ORDINAL,
+	AC_ACCESS_MEMORY_ORDINAL,
+	VIRT_PRIV_ORDINAL,
+	DMI_SERCS_ORDINAL,
+	DMI_SERTX_ORDINAL,
+	DMI_SERRX_ORDINAL
+};
```

Comment:
Here I would like to (politely) restate my opinion that creating this enumeration is incorrect from architectural and API point of view -- for these reasons:

- Mixing of different classes of registers (JTAG, DM, CSR) in one enumeration and in one function (`get_riscv_debug_reg_info()`) is principally incorrect.
- The registers already have their enumerations or indexes defined, so defining another one is redundant and brings the need to convert between them.

Ideally, the API would use the existing indexes and either have one function per the register class (JTAG, DM or CSR); or one function with an additional argument to distinguish the register class. If there are special cases (like tdata1 or textra*), they should be handled by special means (e.g. specific function for those cases) and they IMO do not justify the creation of the enum mix.

Since https://github.com/riscv/riscv-debug-spec/pull/858 has already been reviewed & merged, I respect the decision to leave the current code as is. (No code change asked for here.)